src/core/promgr/promgr.go

```diff
@@ -110,37 +121,40 @@ func (d *defaultProjectManager) Delete(projectIDOrName interface{}) error {
 }
 
 func (d *defaultProjectManager) Update(projectIDOrName interface{}, project *models.Project) error {
-	if len(project.Metadata) > 0 && d.metaMgrEnabled {
-		pro, err := d.Get(projectIDOrName)
-		if err != nil {
+	pro, err := d.Get(projectIDOrName)
+	if err != nil {
+		return err
+	}
+	if pro == nil {
+		return fmt.Errorf("project %v not found", projectIDOrName)
+	}
+	// TODO transaction?
+	if d.metaMgrEnabled {
```

Comment:
check with-clair?

Reply:
I don't think we have to do that for CVE whitelist management as it does not depend on Clair.  The API can be called to manage the whitelist without Clair and once Clair is enabled the whitelist will take effect.

For UX consistency, we may only need to hide certain components from UI if Clair is not installed.